lib/polyfill/patchedmediakeys_20140218.js

```diff
@@ -0,0 +1,800 @@
+/**
+ * Handler for the native keyerror event on MSMediaKeySession.
+ *
+ * @this {!MediaKeySession}
+ * @param {!MediaKeyEvent} event
+ * @private
+ */
+shaka.polyfill.PatchedMediaKeys.v20140218.onMsKeyError_ = function(event) {
+  shaka.log.debug('v20140218.onMsKeyError_', event);
+
+  // Alias
+  var v20140218 = shaka.polyfill.PatchedMediaKeys.v20140218;
+
+  var error = new Error('EME v20140218 key error');
+  error.errorCode = this.nativeMediaKeySession_.error;
+
+  if (this.generateRequestPromise_ != null) {
+    this.generateRequestPromise_.reject(error);
+    this.generateRequestPromise_ = null;
+  }
+  else if (this.updatePromise_ != null) {
+    this.updatePromise_.reject(error);
+    this.updatePromise_ = null;
+  }
+  else {
+    /*
+        Unexpected error - map native codes to standardised key statuses.
+        Possible values of this.nativeMediaKeySession_.error.code
+
+        MS_MEDIA_KEYERR_UNKNOWN        = 1
+        MS_MEDIA_KEYERR_CLIENT         = 2
+        MS_MEDIA_KEYERR_SERVICE        = 3
+        MS_MEDIA_KEYERR_OUTPUT         = 4
+        MS_MEDIA_KEYERR_HARDWARECHANGE = 5
+        MS_MEDIA_KEYERR_DOMAIN         = 6
+        */
+
+    switch (this.nativeMediaKeySession_.error.code) {
+      case MSMediaKeyError.MS_MEDIA_KEYERR_OUTPUT:
+      case MSMediaKeyError.MS_MEDIA_KEYERR_HARDWARECHANGE:
+        this.updateKeyStatus_('output-not-allowed');
+      default:
+        this.updateKeyStatus_('internal-error');
+    }
+  }
+};
+
+
+/**
+ * Update key status and dispatch a 'keystatuseschange' event.
+ *
+ * @param {string} status
+ * @private
+ */
+shaka.polyfill.PatchedMediaKeys.v20140218.MediaKeySession.prototype.
+    updateKeyStatus_ = function(status) {
+  this.keyStatuses.setStatus(status);
+  var event = shaka.util.FakeEvent.create({type: 'keystatuseschange'});
+  this.dispatchEvent(event);
+};
+
+
+
+/**
+ * An implementation of Iterator.
+ *
+ * @param {!Array.<VALUE>} values
+ *
+ * @constructor
+ * @implements {Iterator}
+ * @template VALUE
+ */
+shaka.polyfill.PatchedMediaKeys.v20140218.Iterator = function(values) {
+  /** @private {!Array.<VALUE>} */
+  this.values_ = values;
+
+  /** @private {number} */
+  this.index_ = 0;
+};
+
+
+/**
+ * @return {{value:VALUE, done:boolean}}
+ */
+shaka.polyfill.PatchedMediaKeys.v20140218.Iterator.prototype.next = function() {
+  if (this.index_ >= this.values_.length) {
+    return {value: undefined, done: true};
+  }
+  return {value: this.values_[this.index_++], done: false};
+};
+
+
+
+/**
+ * An implementation of MediaKeyStatusMap.
+ * This fakes a map with a single key ID.
+ *
+ * @constructor
+ * @implements {MediaKeyStatusMap}
+ */
+shaka.polyfill.PatchedMediaKeys.v20140218.MediaKeyStatusMap = function() {
+  /**
+   * @type {number}
+   */
+  this.size = 0;
+
+  /**
+   * @private {string|undefined}
+   */
+  this.status_ = undefined;
+};
+
+
+/**
+ * @const {!Uint8Array}
+ * @private
+ */
+shaka.polyfill.PatchedMediaKeys.v20140218.MediaKeyStatusMap.KEY_ID_;
```

Comment:
It's assigned in the install() func. This was basically copied from the patchedmediakeys_v01b implementation.